### PR TITLE
feat(components): add onValidationError prop to UploadInput

### PR DESCRIPTION
**Pull request by matebek-tw**
## 🖼 Context

The host application already has to provide the `onFileUpload` callback to handle file uploads where it can handle server errors, but in case the file validation fails before the upload callback is triggered, the host application has no way to know about the error at the moment.

## 🚀 Changes

- Add `onValidationError ` prop to `UploadInput`

## ✅ Checklist

- [x] Make PR title meaningful and follow [the commit lint format](https://github.com/transferwise/neptune-web/blob/master/CONTRIBUTING.md#versioning-and-commit-lint) (it will appear in the changelog)
- [x] Changes are tested and all tests pass
- [ ] Changes meet [accessibility standards](https://github.com/transferwise/neptune-web/blob/main/ACCESSIBILITY.md) and there are no violations in the console
- [ ] Changes work in all supported browsers (don't forget IE11)
- [ ] You've updated the documentation if necessary

### CODE REVIEW

File: packages/components/src/uploadInput/UploadInput.tsx [hunk 1]
```diff
@@ -138,31 +147,37 @@ const UploadInput = ({
...
-            },
-          ]);
+          const failedUpload = {
+            id,
+            filename: name,
+            status: Status.FAILED,
+            error: formatMessage(MESSAGES.fileTypeNotSupported),
+          };
+
+          setUploadedFiles((currentFiles) => [...currentFiles, failedUpload]);
```
Comment (StevePole) [inline packages/components/src/uploadInput/UploadInput.tsx:157]: Should we append files that file validation?

Comment (matebek-tw) [reply packages/components/src/uploadInput/UploadInput.tsx:157]: We are appending them to the UI to display them in a failed state so the user will see the failed uploads and the error message.

File: packages/components/src/uploadInput/UploadInput.tsx [hunk 2]
```diff
@@ -41,6 +41,12 @@ export type UploadInputProps = {
    */
   onDeleteFile?: (id: string | number) => Promise<any>;
 
+  /**
+   * Provide a callback to trigger on validation error
+   * @param file
+   */
+  onValidationError?: (file: UploadedFile) => void;
```
Comment (dozortsev) [inline packages/components/src/uploadInput/UploadInput.tsx:48]: I would probably call this a bit differently, e.g `onFailedUpload`, I think this callback could be triggered not only for validation failure reasons

**Review by dozortsev — APPROVED** [review]

File: packages/components/src/uploadInput/UploadInput.tsx (hunk 2 above)
Comment (matebek-tw) [reply packages/components/src/uploadInput/UploadInput.tsx:48]: The `onUploadFile` callback is already provided by the consumer app, where they can handle failed uploads. But that callback wouldn't be called at all when any of the validations are failing, hence the `onValidationError` API.

Comment (dozortsev) [reply packages/components/src/uploadInput/UploadInput.tsx:48]: I see, make sense 👍 

could you please mention that consumers can handle failed uploads in JSDoc for that prop?

**Pull request merged**